src/components/shared/Entity/EndpointConfig/index.tsx

```diff
@@ -55,17 +51,36 @@ function EndpointConfig({
     // Endpoint Config Store
     const endpointConfig = useEndpointConfigStore_endpointConfig_data();
     const setEndpointConfig = useEndpointConfigStore_setEndpointConfig();
-
     const previousEndpointConfig =
         useEndpointConfigStore_previousEndpointConfig_data();
     const setPreviousEndpointConfig =
         useEndpointConfigStore_setPreviousEndpointConfig();
-
     const endpointSchema = useEndpointConfigStore_endpointSchema();
     const setEndpointSchema = useEndpointConfigStore_setEndpointSchema();
-
     const setServerUpdateRequired = useEndpointConfig_setServerUpdateRequired();
+    const setEndpointCanBeBlank = useEndpointConfig_setEndpointCanBeBlank();
+
+    // Workflow related props
+    const workflow = useEntityWorkflow();
+    const editWorkflow =
+        workflow === 'capture_edit' || workflow === 'materialization_edit';
+    const forceClose = !editWorkflow && draftId !== null;
```

Comment:
Moved props so they are grouped together to feel out a slightly different pattern